<?xml version="1.0" ?>

<launch>
  <arg name="load_gripper" default="true" />
  <arg name="robot_ip" />
  <arg name="publish_rate" default="30" />

  <param name="robot_description" command="$(find xacro)/xacro $(find franka_description)/robots/panda_arm.urdf.xacro hand:=$(arg load_gripper)" />

  <node name="robot_joint_state_publisher" pkg="franka_visualization" type="robot_joint_state_publisher" output="screen">
    <rosparam command="load" file="$(find franka_visualization)/config/robot_settings.yaml" />
    <param name="robot_ip" value="$(arg robot_ip)" />
    <param name="publish_rate" value="$(arg publish_rate)" />
    <remap from="~/joint_states" to="joint_states" />
  </node>

  <node name="gripper_joint_state_publisher" pkg="franka_visualization" type="gripper_joint_state_publisher" output="screen" if="$(arg load_gripper)">
    <rosparam command="load" file="$(find franka_visualization)/config/gripper_settings.yaml" />
    <param name="robot_ip" value="$(arg robot_ip)" />
    <param name="publish_rate" value="$(arg publish_rate)" />
    <remap from="~/joint_states" to="joint_states" />
  </node>

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
  <node pkg="rviz" type="rviz" output="screen" name="rviz" args="-d $(find franka_visualization)/launch/franka_visualization.rviz"/>
</launch>
